extensions/in-memory/assetindex-memory/src/test/java/org/eclipse/dataspaceconnector/metadata/memory/InMemoryDataAddressResolverTest.java

```diff
@@ -39,6 +40,17 @@ void resolveForAsset_assetNull_raisesException() {
         assertThatThrownBy(() -> resolver.resolveForAsset(null)).isInstanceOf(NullPointerException.class);
     }
 
+    @Test
+    void resolveForAsset_whenAssetDeleted_raisesException() {
+        var testAsset = createAsset("foobar", UUID.randomUUID().toString());
+        var address = createDataAddress(testAsset);
+        resolver.accept(testAsset, address);
+        resolver.deleteById(testAsset.getId());
+        assertThatThrownBy(() -> resolver.resolveForAsset(testAsset.getId()))
+                .hasMessage(String.format("No DataAddress found for Asset ID=%s", testAsset.getId()))
+                .isInstanceOf(IllegalArgumentException.class);
```

Comment:
nit: consider putting `isInstanceOf` before `hasMessage ` so that the problem will be clearer if a different exception is thrown.